test/test_utils.py

```diff
@@ -611,6 +611,28 @@ def test_timer(self):
         median = timer.blocked_autorange(min_run_time=0.1).median
         self.assertIsInstance(median, float)
 
+    def test_adaptive_timer(self):
```

Comment:
Can you also test that the results of `adaptive_autorange` return False for `has_warnings`?

Reply:
done.